tests/integ/test_simplifier_suite.py

```diff
@@ -1196,17 +1225,23 @@ def test_chained_sort(session):
         # Not flattened, since we cannot flatten when the subquery uses positional parameter ($1)
         (
             lambda df: df.filter(col("$1") > 1).select(col("B"), col("A")),
-            'SELECT "B", "A" FROM ( SELECT  *  FROM ( SELECT "A", "B" FROM ( SELECT $1 AS "A", $2 AS "B" FROM  VALUES (1 :: INT, -2 :: INT), (3 :: INT, -4 :: INT))) WHERE ("$1" > 1 :: INT))',
+            'SELECT "B", "A" FROM ( SELECT  *  FROM ( SELECT "A", "B" FROM ( SELECT $1 AS "A", $2 AS "B" FROM  VALUES (1 :: INT, -2 :: INT), (3 :: INT, -4 :: INT))) WHERE ("$1" > 1$POSTFIX))',
         ),
     ],
 )
-def test_select_after_filter(session, operation, simplified_query):
+def test_select_after_filter(setup_reduce_cast, session, operation, simplified_query):
     session.sql_simplifier_enabled = False
     df1 = session.create_dataframe([[1, -2], [3, -4]], schema=["a", "b"])
 
     session.sql_simplifier_enabled = True
     df2 = session.create_dataframe([[1, -2], [3, -4]], schema=["a", "b"])
 
+    # replace '$POSTFIX' with the expected integer literal postfix to get the final expected simplified query
+    integer_literal_postfix = (
+        "" if session.eliminate_numeric_sql_value_cast_enabled else " :: INT"
+    )
+    simplified_query = simplified_query.replace("$POSTFIX", integer_literal_postfix)
```

Comment:
you can use https://docs.python.org/3/library/stdtypes.html#str.format instead of replacing strings

Reply:
updated